pkg/sdn/registry/netnamespace/rest.go

```diff
@@ -0,0 +1,243 @@
+package netnamespace
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/golang/glog"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/fielderrors"
+	"k8s.io/kubernetes/pkg/watch"
+
+	"github.com/openshift/openshift-sdn/plugins/osdn/util"
+	"github.com/openshift/openshift-sdn/plugins/osdn/vnidallocator"
+	"github.com/openshift/origin/pkg/sdn/api"
+	"github.com/openshift/origin/pkg/sdn/api/validation"
+	"github.com/openshift/origin/pkg/sdn/registry/netnamespace/cache"
+)
+
+// REST adapts a netnamespace registry into apiserver's RESTStorage model.
+type REST struct {
+	registry Registry
+	vnids    vnidallocator.Interface
+	// globalNamespaces are the namespaces that have access to all pods in the cluster and vice versa.
+	globalNamespaces []string
```

Comment:
make this a sets.String so you can do O(1) lookups with `globalNamespaces.Has(ns)`